src/Http/Client.php

```diff
@@ -100,11 +102,18 @@
  * a proxy if you need to use one. The type sub option can be used to
  * specify which authentication strategy you want to use.
  * CakePHP comes with built-in support for basic authentication.
+ *
+ * @implements \Cake\Event\EventDispatcherInterface<\Cake\Http\Client>
  */
-class Client implements ClientInterface
+class Client implements ClientInterface, EventDispatcherInterface
```

Comment:
With this change composer.json of the http package needs to be updated to add the event package as a dependency.

Reply:
[it already does](https://github.com/cakephp/cakephp/blob/5.x/src/Http/composer.json#L30)

Reply:
Ah right 👍